src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -566,23 +566,39 @@ trait Implicits {
      }
 
     private def matchesPtInst(info: ImplicitInfo): Boolean = {
+      if (StatisticsStatics.areSomeColdStatsEnabled) statistics.incCounter(matchesPtInstCalls)
+
       def isViewLike = pt match {
         case Function1(_, _) => true
         case _ => false
       }
 
-      if (StatisticsStatics.areSomeColdStatsEnabled) statistics.incCounter(matchesPtInstCalls)
-        info.tpe match {
-          case PolyType(tparams, restpe) =>
-            try {
-              val allUndetparams = (undetParams ++ tparams).distinct
-              val tvars = allUndetparams map freshVar
-              val tp = ApproximateDependentMap(restpe)
-              val tpInstantiated = tp.instantiateTypeParams(allUndetparams, tvars)
+      info.tpe match {
+        case PolyType(tparams, restpe) =>
+          try {
+            val allUndetparams = (undetParams ++ tparams).distinct
+            val tvars = allUndetparams map freshVar
+            val tp = ApproximateDependentMap(restpe)
+            val tpInstantiated = tp.instantiateTypeParams(allUndetparams, tvars)
+
+            if(isView || isViewLike) {
+              tpInstantiated match {
+                case MethodType(_, tv: TypeVar) if !tv.instValid =>
```

Comment:
I think this check could be more general. What if it's a conversion like `def foo(x: From): x.To`? The result type would also be undetermined, right? Or it could be an intersection type of a type variable with some other type.

Could you add a few test cases along these lines and make sure we catch those as well?